packages/backend/src/tokens/__tests__/machine.test.ts

```diff
@@ -0,0 +1,55 @@
+import { describe, expect, it } from 'vitest';
+
+import { API_KEY_PREFIX, getMachineTokenType, isMachineToken, M2M_TOKEN_PREFIX, OAUTH_TOKEN_PREFIX } from '../machine';
+
+describe('isMachineToken', () => {
+  it('returns true for tokens with M2M prefix', () => {
+    expect(isMachineToken(`${M2M_TOKEN_PREFIX}some-token-value`)).toBe(true);
+  });
+
+  it('returns true for tokens with OAuth prefix', () => {
+    expect(isMachineToken(`${OAUTH_TOKEN_PREFIX}some-token-value`)).toBe(true);
+  });
+
+  it('returns true for tokens with API key prefix', () => {
+    expect(isMachineToken(`${API_KEY_PREFIX}some-token-value`)).toBe(true);
+  });
+
+  it('returns false for tokens without a recognized prefix', () => {
+    expect(isMachineToken('unknown_prefix_token')).toBe(false);
```

Comment:
Just wanna note that we do plan to allow custom prefixes in the future - likely these end up being prepended to the token type prefix so i think it should be a fairly straightforward change

Reply:
Added a note to cover this in the future.